src/linalg/dot.cairo

```diff
@@ -24,11 +25,14 @@ fn dot<
     let mut index = 0;
     let mut value = Zeroable::zero();
     loop {
-        if index == xs.len() {
-            break ();
-        }
-        value += *xs[index] * *ys[index];
-        index += 1;
-    };
-    value
+        match xs.pop_front() {
+            Option::Some(x_value) => {
+                let y_value = ys.pop_front().unwrap();
+                value += *x_value * *y_value;
+            },
+            Option::None(_) => {
+                break value;
```

Comment:
why do you return value ?

Reply:
To return the "value" as it was existing before.
Since loop isn't ending by a `;` it'll return that break value as the return of the fn